File: software/examples/csharp/ExampleCallback.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Color Bricklet 2.0

	// Callback function for color callback
	static void ColorCB(BrickletColorV2 sender, int r, int g, int b, int c)
	{
		Console.WriteLine("Color [R]: " + r);
		Console.WriteLine("Color [G]: " + g);
		Console.WriteLine("Color [B]: " + b);
		Console.WriteLine("Color [C]: " + c);
		Console.WriteLine("");
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletColorV2 c = new BrickletColorV2(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Register color callback to function ColorCB
		c.ColorCallback += ColorCB;

		// Set period for color callback to 0.1s (100ms)
		c.SetColorCallbackConfiguration(100, false);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/perl/example_callback.pl
#!/usr/bin/perl

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickletColorV2;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your Color Bricklet 2.0

# Callback subroutine for color callback
sub cb_color
{
    my ($r, $g, $b, $c) = @_;

    print "Color [R]: $r\n";
    print "Color [G]: $g\n";
    print "Color [B]: $b\n";
    print "Color [C]: $c\n";
    print "\n";
}

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $c = Tinkerforge::BrickletColorV2->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Register color callback to subroutine cb_color
$c->register_callback($c->CALLBACK_COLOR, 'cb_color');

# Set period for color callback to 0.1s (100ms)
$c->set_color_callback_configuration(100, 0);

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();
